### ISSUE TITLE
Add `MirrorReadableKVState` implementation for `ReadableKVState`

### ISSUE THREAD

**Issue by IvanKavaldzhiev** [hashgraph/hedera-mirror-node#9250]
### Problem

In order to integrate the reusable services internal state handling with the DB layer, we should add mirror node specific implementation of the `ReadableKVState` interface that will hold map of specific ReadableKVState for each service. For example:

The `MirrorReadableKVState` for tokens will contain:
- AccountReadableKVState
- TokenReadableKVState
- TokenRelationshipReadableKVState
etc..

### Solution

Add `MirrorReadableKVState` implementation, which and have a caching mechanism to keep read only access of entities that are already fetched from DB. This cache might make having a cache inside the specific ReadableKVState for a given type obsolete.

Perform unit and integration tests for all initialization types of `MirrorReadableKVState`, reading all types of data from DB, that will be needed. That is:

- contract storage
- contract bytecode
- accounts
- tokens ( + nfts)
- aliases
- token relationships
- airdrops
- files

Note: this task is blocked by the tasks for adding the needed ReadableKVState implementations.

**Comment by IvanKavaldzhiev** [comment]
Closing in favour of #9259
